target/linux/generic/pending-5.10/999-6-net-dsa-qca8k-cache-lo-and-hi-for-mdio-write.patch

```diff
@@ -0,0 +1,85 @@
+From b2f39636fbb951bc60ec4d3cd77cd878304554e0 Mon Sep 17 00:00:00 2001
+From: Ansuel Smith <ansuelsmth@gmail.com>
+Date: Tue, 7 Dec 2021 04:43:06 +0100
+Subject: [PATCH 6/6] net: dsa: qca8k: cache lo and hi for mdio write
+
+From Documentation, we can cache lo and hi the same way we do with the
+page. This massively reduce the mdio write as 3/4 of the time we only
+require to write the lo or hi part for a mdio write.
+
+Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
+---
+ drivers/net/dsa/qca8k.c | 49 ++++++++++++++++++++++++++++++++++++-----
+ 1 file changed, 44 insertions(+), 5 deletions(-)
+
+diff --git a/drivers/net/dsa/qca8k.c b/drivers/net/dsa/qca8k.c
+index d2c6139be9ac..64643f1e2f16 100644
+--- a/drivers/net/dsa/qca8k.c
++++ b/drivers/net/dsa/qca8k.c
+@@ -94,6 +94,48 @@ qca8k_split_addr(u32 regaddr, u16 *r1, u16 *r2, u16 *page)
+ 	*page = regaddr & 0x3ff;
+ }
+ 
++static u16 qca8k_current_lo = 0xffff;
```

Comment:
We should consider a more safe value.... But considering the first thing will be checking the switch id... this should be fine...